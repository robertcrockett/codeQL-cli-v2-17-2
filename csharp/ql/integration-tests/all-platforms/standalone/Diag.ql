import csharp
import semmle.code.csharp.commons.Diagnostics

query predicate extractorMessages(int c) { c = count(ExtractorMessage msg) }

query predicate compilerDiagnostics(int c) { c = count(Diagnostic diag) }

query predicate compilationInfo(string key, float value) {
  exists(Compilation c, string infoValue |
    infoValue = c.getInfo(key) and key.matches("Compiler diagnostic count for%")
  |
    value = infoValue.toFloat()
  )
}
