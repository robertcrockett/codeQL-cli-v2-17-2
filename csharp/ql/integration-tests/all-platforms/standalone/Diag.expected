extractorMessages
| 5 |
compilerDiagnostics
| 4 |
compilationInfo
| Compiler diagnostic count for CS0103 | 3.0 |
| Compiler diagnostic count for CS8019 | 7.0 |
